File: backend/app/schemas/maquina_gym.py
from datetime import datetime

from pydantic import BaseModel


class MaquinaGymCreate(BaseModel):
    nombre: str
    grupo_muscular: str | None = None
    descripcion_uso: str | None = None


class MaquinaGymUpdate(BaseModel):
    nombre: str | None = None
    grupo_muscular: str | None = None
    descripcion_uso: str | None = None


class MaquinaGymResponse(BaseModel):
    id: str
    usuario_id: str
    nombre: str
    foto_path: str | None
    descripcion_uso: str | None
    grupo_muscular: str | None
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: flutter_app/lib/features/machines/domain/maquina_gym.dart
import '../../../core/app_config.dart';
import '../../../core/env.dart';

class MaquinaGym {
  final String id;
  final String usuarioId;
  final String nombre;
  final String? fotoPath;
  final String? descripcionUso;
  final String? grupoMuscular;
  final DateTime createdAt;
  final DateTime updatedAt;

  String? get imageUrl {
    if (fotoPath == null) return null;
    final activeBaseUrl = AppConfig.baseUrl ?? Env.apiBaseUrl;
    final baseUrl = activeBaseUrl.replaceAll('/api/v1', '');
    final cleanPath = fotoPath!.replaceAll('backend/', '').replaceAll('backend/storage/', 'storage/');
    final path = cleanPath.startsWith('/') ? cleanPath : '/$cleanPath';
    if (path.contains('/api/v1/')) {
      return '$baseUrl$path';
    } else {
      final formattedPath = path.replaceAll('/storage/', '/api/v1/storage/');
      return '$baseUrl$formattedPath';
    }
  }

  MaquinaGym({
    required this.id,
    required this.usuarioId,
    required this.nombre,
    this.fotoPath,
    this.descripcionUso,
    this.grupoMuscular,
    required this.createdAt,
    required this.updatedAt,
  });

  factory MaquinaGym.fromJson(Map<String, dynamic> json) {
    return MaquinaGym(
      id: json['id'] as String,
      usuarioId: json['usuario_id'] as String,
      nombre: json['nombre'] as String,
      fotoPath: json['foto_path'] as String?,
      descripcionUso: json['descripcion_uso'] as String?,
      grupoMuscular: json['grupo_muscular'] as String?,
      createdAt: DateTime.parse(json['created_at'] as String),
      updatedAt: DateTime.parse(json['updated_at'] as String),
    );
  }
}